src/views/docs-view/utils/__tests__/all-docs-server.test.ts

```diff
@@ -0,0 +1,104 @@
+import { RootDocsPath } from 'types/products'
+import {
+	configureRootDocsPathParams,
+	removeCustomLandingPaths,
+	parseRootDocsPath,
+} from '../all-docs-server'
+
+describe('configureRootDocsPathParams', () => {
+	it('prefixes params with a provided basePath, and renames param to allDocs', () => {
+		// Input
+		const paths = [
+			{
+				params: {
+					page: ['foo', 'bar'],
+				},
+			},
+		]
+		const basePath = 'some-base-path'
+		// Expected output
+		const expected = [
+			{
+				params: {
+					rootDocsPath: 'some-base-path',
+					docsSlug: ['foo', 'bar'],
+				},
+			},
+		]
+		expect(configureRootDocsPathParams(paths, basePath)).toEqual(expected)
+	})
+})
+
+describe('removeCustomLandingPaths', () => {
+	it('removes custom landing paths', () => {
+		// Input
+		const paths = [
+			['some-base-path'],
+			['some-base-path', 'foo'],
+			['some-base-path', 'foo', 'bar'],
+			['another-base-path'],
+			['another-base-path', 'fizz'],
+		].map((params) => {
+			return { params: { allDocs: params } }
+		})
+		const customLandingPaths = ['some-base-path']
+		// Expected output
+		const expected = [
+			['some-base-path', 'foo'],
+			['some-base-path', 'foo', 'bar'],
+			['another-base-path'],
+			['another-base-path', 'fizz'],
+		].map((params) => {
+			return { params: { allDocs: params } }
+		})
+		expect(removeCustomLandingPaths(paths, customLandingPaths)).toEqual(
+			expected
+		)
```

Comment:
You can use [inline-snapshots](https://jestjs.io/docs/snapshot-testing#inline-snapshots) to avoid hard coding the _expected outputs_!

```suggestion
		expect(removeCustomLandingPaths(paths, customLandingPaths)). toMatchInlineSnapshot()
```